src/dials/algorithms/symmetry/cosym/__init__.py

```diff
@@ -831,3 +873,35 @@ def as_dict(self):
             "cb_op": f"{self.subgroup['cb_op_inp_best']}",
             "stars": self.stars,
         }
+
+
+def extract_reference_intensities(params: iotbx.phil.scope_extract) -> miller.array:
+    # Extract/calculate a set of intensities from a reference.
+    if params.d_min not in {Auto, None}:
+        reference_intensities = intensities_from_reference_file(
+            params.reference, d_min=params.d_min
+        )
+    else:
+        reference_intensities = intensities_from_reference_file(params.reference)
+    initial_space_group_info = reference_intensities.space_group_info()
+    group = metric_subgroups(
+        reference_intensities.crystal_symmetry(),
+        params.lattice_symmetry_max_delta,
+        enforce_max_delta_for_generated_two_folds=True,
+    ).result_groups[0]
+    ref_cb_op = (
+        group["best_subsym"].change_of_basis_op_to_minimum_cell()
+        * group["cb_op_inp_best"]
+    )
+
+    with open(os.devnull, "w") as devnull:
+        reference_intensities, _ = reference_intensities.apply_change_of_basis(
+            str(ref_cb_op), out=devnull
+        )
+    reference_intensities = reference_intensities.expand_to_p1()
+    reference_intensities = (
+        reference_intensities.as_non_anomalous_array().merge_equivalents().array()
+    )
```

Comment:
If you use `miller.array.change_basis()` rather than `miller.array.apply_change_of_basis()` then you don't need the `os.devnull` dance:
```suggestion
    reference_intensities = (
        reference_intensities.change_basis(
            ref_cb_op,
        )
        .expand_to_p1()
        .as_non_anomalous_array()
        .merge_equivalents()
        .array()
    )
```